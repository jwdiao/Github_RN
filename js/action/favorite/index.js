import Types from '../types'
import DataStore, {FLAG_STORAGE} from '../../expand/dao/DataStore'
import {handleData, _projectModels} from '../ActionUtil'
import FavoriteDao from '../../expand/dao/FavoriteDao'
import ProjectModel from '../../model/ProjectModel'

export function onLoadFavoriteData(flag, isShowLoading) {
  return dispatch => {
    dispatch({
      type: Types.FAVORITE_LOAD_DATA,
      storeName: flag
    })
    new FavoriteDao(flag).getAllItems()
        .then(items => {
          let resultData = []
          for (let i = 0; i < items.length; i++) {
            resultData.push(new ProjectModel(items[i], true))
          }
          dispatch({
            type: Types.FAVORITE_LOAD_SUCCESS,
            projectModels: resultData,
            storeName: flag
          })
        }).catch(e => {
            console.log(e)
            dispatch({
              type: Types.FAVORITE_LOAD_FAIL,
              storeName: flag,
              error: e
            })
        })
  }
}
