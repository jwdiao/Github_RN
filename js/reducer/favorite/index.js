import Types from '../../action/types'

/*
   favorite: {
      popular: {
         projectModels: [],
         isLoading: false
      },
      trending: {
         projectModels: [],
         isLoading: false
      }
   }
*/
export const defaultState = {}

export default function onAction(state = defaultState, action) {
  switch (action.type) {
    case Types.FAVORITE_LOAD_DATA:
      return {
        ...state,
        [action.storeName]: {
          ...state[action.storeName],
          isLoading: true,
        }
      }
    case Types.FAVORITE_LOAD_SUCCESS:
       return {
         ...state,
         [action.storeName]: {
           ...state[action.storeName],
           isLoading: false,
           projectModels: action.projectModels
         }
       }
    case Types.FAVORITE_LOAD_FAIL:
       return {
         ...state,
         [action.storeName]: {
           ...state[action.storeName],
           isLoading: false
         }
       }
    default:
    return state
  }
}
