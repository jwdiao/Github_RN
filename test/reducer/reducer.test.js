import {onLoadFavoriteData} from '../../js/action/favorite/index'
import onAction, {defaultState} from '../../js/reducer/favorite/index'
import Types from '../../js/action/types'

it('test reducer',() => {
  const nextState = onAction(defaultState, {
    type: Types.FAVORITE_LOAD_DATA,
    storeName: 'keys'
  })
  expect(nextState).toMatchSnapshot()
})
