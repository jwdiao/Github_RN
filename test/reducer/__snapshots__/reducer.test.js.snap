// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test reducer 1`] = `
Object {
  "keys": Object {
    "isLoading": true,
  },
}
`;
